doc/Programming-guide/Your-first-Kaa-application/index.md

```diff
@@ -422,14 +422,162 @@ int main(void)
+ */
+public class FirstKaaDemo {
+
+    private static final long DEFAULT_START_DELAY = 1000L;
+
+    private static final Logger LOG = LoggerFactory.getLogger(FirstKaaDemo.class);
+
+    private static KaaClient kaaClient;
+
+    private static ScheduledFuture<?> scheduledFuture;
+    private static ScheduledExecutorService scheduledExecutorService;
+
+    public static void main(String[] args) {
+        LOG.info(FirstKaaDemo.class.getSimpleName() + " app starting!");
+
+        scheduledExecutorService = Executors.newScheduledThreadPool(1);
+        /*
+         * Create the Kaa desktop context for the application.
+         */
+        DesktopKaaPlatformContext desktopKaaPlatformContext = new DesktopKaaPlatformContext();
+
+        /*
+         * Create a Kaa client and add a listener which displays the Kaa client
+         * configuration as soon as the Kaa client is started.
+         */
+        kaaClient = Kaa.newClient(desktopKaaPlatformContext, new FirstKaaClientStateListener(), true);
+
+        /*
+         *  Used by log collector on each adding of the new log record in order to check whether to send logs to server.
+         *  Start log upload when there is at least one record in storage.
+         */
+        RecordCountLogUploadStrategy strategy = new RecordCountLogUploadStrategy(1);
+        strategy.setMaxParallelUploads(1);
+        kaaClient.setLogUploadStrategy(strategy);
+
+        /*
+         * Persist configuration in a local storage to avoid downloading it each
+         * time the Kaa client is started.
+         */
+        kaaClient.setConfigurationStorage(new SimpleConfigurationStorage(desktopKaaPlatformContext, "saved_config.cfg"));
+
+        kaaClient.addConfigurationListener(new ConfigurationListener() {
+            @Override
+            public void onConfigurationUpdate(Configuration configuration) {
+                LOG.info("Received configuration data. New sample period: {}", configuration.getSamplePeriod());
+                onChangedConfiguration(TimeUnit.SECONDS.toMillis(configuration.getSamplePeriod()));
+            }
+        });
+
+        /*
+         * Start the Kaa client and connect it to the Kaa server.
+         */
+        kaaClient.start();
+
+        LOG.info("--= Press any key to exit =--");
+        try {
+            System.in.read();
+        } catch (IOException e) {
+            LOG.error("IOException has occurred: {}", e.getMessage());
+        }
+        LOG.info("Stopping...");
 
-</div>
+        scheduledExecutorService.shutdown();
+        kaaClient.stop();
+    }
 
-<div id="app-obj-c" class="tab-pane fade" markdown="1" >
+    /*
+    * Method, that emulate getting temperature from real sensor.
+    * Retrieves random temperature.
+    */
+    private static int getTemperatureRand() {
+        return new Random().nextInt(10) + 31;
+    }
 
-</div>
+    private static void onKaaStarted(long time) {
+        if (time == 0) {
```

Comment:
This check is useless. Default configuration has "by_default": "1". Better check for negative values.